C++/4. Advanced-Operators.md

```diff
@@ -66,46 +67,49 @@ Relational Operators are operators which compare things together:
 - `==`
 - `!=`
 
-`<` means less than. \
-`>` means greater than. \
-`<=` means less than or equal to. \
-`>=` means greater than or equal to. \
-`==` means equal to. \
-`!=` means not equal to.
+`<` means "less than." \
+`>` means "greater than." \
+`<=` means "less than or equal to." \
+`>=` means "greater than or equal to." \
+`==` means "equal to." \
+`!=` means "not equal to."
 
-All of the above operators check the conditions. Like `==` checks if they are equal.
+All of the above operators compare two values against each other. For example, `x == y` checks if `x` and `y` are equal.
+
+**Note: At the lowest level, computers represent booleans as an integer, '0' or '1' (for "false" and "true" respectively), and normally C++'s `std::cout <<` will quite literally show `0` or `1`. We can change this behavior by manipulating `std::cout`, by using `std::boolalpha`, which will now show `true` or `false`, instead of `1` or `0`. (`std::cout` is a `std::ostream`, streams will be covered much later)**
 
 ### Code
 
 ```cpp
 #include <iostream>
 
 int main() {
-  bool boolean1 = 4 < 3;
-  bool boolean2 = 4 > 3;
-  bool boolean3 = 4 <= 3;
-  bool boolean4 = 4 >= 3;
-  bool boolean5 = 4 == 3;
-  bool boolean6 = 4 != 3;
-
-  std::cout << boolean1 << std::endl
-            << boolean2 << std::endl
-            << boolean3 << std::endl
-            << boolean4 << std::endl
-            << boolean5 << std::endl
-            << boolean6 << std::endl;
+  bool four_is_less_than_three = 4 < 3;
+  bool four_is_greater_than_three = 4 > 3;
+  bool four_is_less_than_or_equal_to_three = 4 <= 3;
+  bool four_is_greater_than_or_equal_to_three = 4 >= 3;
+  bool four_is_three = 4 == 3;
+  bool four_is_not_three = 4 != 3;
```

Comment:
too big names and are bad. This is bad code quality. Don't put the names soo exact - just keep it as integer1 or something

Reply:
Okay, how about:
```c++
bool less_than = 4 < 3;
bool greater_than = 4 > 3;
bool less_than_or_equal = 4 <= 3;
bool greater_than_or_equal = 4 >= 3;
bool equal = 4 == 3;
bool not_equal = 4 != 3;
```

Retains semantic naming, without being nonsense bool0, bool1, names, what do you think?

Reply:
`less_than_or_equal` & `greater_than_or_equal` are too long. You can change it to `less_or_equal` & `greater_or_equal` as the message is conveyed.

Also cammelCase seems better than using underscores.

Reply:
IMHO, they aren't long at all, real code has much longer names. But okay, I guess I'll shorten it further?

Also, C++, like C, Python, Rust, etc, is generally considered to be a snake_case language, are you sure that you'd rather use CamelCase in the tutorial instead?

Reply:
anything - your wish. If you think that those languages are considered to be snake_case, then snake_case it is